go-controller/pkg/ovn/ovn.go

```diff
@@ -180,6 +182,16 @@ func (oc *DefaultNetworkController) ensureLocalZonePod(oldPod, pod *kapi.Pod, ad
 		}
 	}
 
+	if kubevirt.IsPodLiveMigratable(pod) {
+		switchNames, err := oc.getSwitchNames(pod)
+		if err != nil {
+			return fmt.Errorf("failed retrieving switch names when ensuring local zone for pod %s/%s: %v", pod.Namespace, pod.Name, err)
+		}
+		if err := kubevirt.EnsureRoutingForVM(oc.controllerName, oc.watchFactory, oc.nbClient, switchNames, true /* local zone */, pod, oc.GetNetworkName()); err != nil {
```

Comment:
Why don't you do this once in `ensurePod`

Reply:
They do different things depending on the zone, problem is that the method name is the same, using a flag to denote that is no good. I am going to refactor this to have a pair of different method so it's more clear.